src/results.cpp

```diff
@@ -696,6 +699,20 @@ NotificationToken Results::add_notification_callback(CollectionChangeCallback cb
     return {m_notifier, m_notifier->add_callback(std::move(cb))};
 }
 
+void Results::subscribe(util::Optional<std::string> subscription_name)
```

Comment:
It bothers me a bit that this is not asynchronous call, considering it's feasible that an error occurs and the subscription doesn't register. Also, it will be asymmetric with regard to unsubscribing, which triggers my OCD. Are there compelling arguments for keeping it synchronous?

Reply:
It is async (register_partial_sync_query() does the actual work on a different thread).

Reply:
Yeah, I meant that it doesn't surface in the public API as such - there's no way for the consumer to get a confirmation that their subscription has been "registered".

Reply:
This is more explicit in #628. Subscribing returns a `Subscription` object that you can observe for state changes / errors.